src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,468 @@
+        origTrack.selected = true;
+        return;
+      }
+    }
+    throw new Error("Video track not found.");
+  }
+
+  public getChosenAudioTrack(): ITMAudioTrack|null|undefined {
+    for (let i = 0; i < this._audioTracks.length; i++) {
+      const { track, origTrack } = this._audioTracks[i];
+      if (origTrack.enabled) {
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getChosenTextTrack(): ITMTextTrack|null|undefined {
+    for (let i = 0; i < this._textTracks.length; i++) {
+      const { track, origTrack } = this._textTracks[i];
+      if (origTrack.mode === "showing") {
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getChosenVideoTrack(): ITMVideoTrack|null|undefined {
+    for (let i = 0; i < this._videoTracks.length; i++) {
+      const { track, origTrack } = this._videoTracks[i];
+      if (origTrack.selected) {
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getAvailableAudioTracks(): ITMAudioTrackListItem[]|undefined {
+    return this._audioTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               language: track.language,
+               normalized: track.normalized,
+               audioDescription: track.audioDescription,
+               active: origTrack.enabled };
+    });
+  }
+
+  public getAvailableTextTracks(): ITMTextTrackListItem[]|undefined {
+    return this._textTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               language: track.language,
+               normalized: track.normalized,
+               closedCaption: track.closedCaption,
+               active: origTrack.mode === "showing" };
+    });
+  }
+
+  public getAvailableVideoTracks(): ITMVideoTrackListItem[]|undefined {
+    return this._videoTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               representations: track.representations,
+               active: origTrack.selected };
+    });
+  }
+
+  public dispose(): void {
+    this._mediaElement.videoTracks.onchange = null;
+    this._mediaElement.videoTracks.onaddtrack = null;
+    this._mediaElement.videoTracks.onremovetrack = null;
+
+    this._mediaElement.audioTracks.onchange = null;
+    this._mediaElement.audioTracks.onaddtrack = null;
+    this._mediaElement.audioTracks.onremovetrack = null;
+
+    (this._mediaElement.textTracks as ICustomTextTrackList).onchange = null;
+    this._mediaElement.textTracks.onaddtrack = null;
+    (this._mediaElement.textTracks as ICustomTextTrackList).onremovetrack = null;
+
+    this.removeEventListener();
+  }
+
+  private _setInitialAudioTrack() : void {
+    const preferredAudioTracks = this._preferredAudioTracks.getValue();
+    const normalizedLanguages = preferredAudioTracks
+      .filter(
+        (audioTrack): audioTrack is {
+          language : string;
+          audioDescription : boolean;
+        } => audioTrack !== null)
+      .map(({ language }) => normalizeLanguage(language));
```

Comment:
So if he wants an audiodescription in a language B or a "normal" in a language A, we take the language B? To check if that's what we want.